core/include/cubos/core/al/miniaudio_device.hpp

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+#include <cstddef>
+
+#include <cubos/core/al/audio_device.hpp>
+
+namespace cubos::core::al
+{
+    class MiniaudioBuffer : public impl::Buffer
+    {
+    public:
+        ~MiniaudioBuffer() override;
+        size_t getLength() override;
+
+        static std::shared_ptr<Buffer> create(const void* data, size_t dataSize)
+        {
+            return std::shared_ptr<cubos::core::al::MiniaudioBuffer>(new MiniaudioBuffer(data, dataSize));
+        }
+
+    protected:
+        MiniaudioBuffer(const void* data, size_t dataSize);
+    };
+
+    class MiniaudioSource : public impl::Source
+    {
+    public:
+        ~MiniaudioSource() override;
+
+        static std::shared_ptr<Source> create()
+        {
+            return std::shared_ptr<cubos::core::al::MiniaudioSource>(new MiniaudioSource());
+        }
+
+        void setBuffer(cubos::core::al::Buffer buffer) override;
+        void setPosition(const glm::vec3& position) override;
+        void setVelocity(const glm::vec3& velocity) override;
+        void setGain(float gain) override;
+        void setPitch(float pitch) override;
+        void setLooping(bool looping) override;
+        void setRelative(bool relative) override;
+        void setMaxDistance(float maxDistance) override;
+        void setMinDistance(float minDistance) override;
+        void setCone(float innerAngle, float outerAngle, float outerGain) override;
+        void setConeDirection(const glm::vec3& direction) override;
+        void play() override;
+
+    protected:
+        MiniaudioSource();
+
+    private:
+        ma_sound mSound;
+        ma_engine mEngine;
+    };
```

Comment:
These classes can be moved to the `miniaudio_device.cpp`, as they're never created manually outside this implementation.